runtime/databricks/automl_runtime/sklearn/base_datetime_transformer.py

```diff
@@ -36,16 +40,69 @@ class BaseDatetimeTransformer(ABC, TransformerMixin, BaseEstimator):
     DAYS_IN_YEAR = 366  # Account for leap years
     WEEKEND_START = 5
 
+    def __init__(self, impute_method=None):
+        """Create a `BaseDatetimeTransformer`.
+
+        Parameters
+        ----------
+        impute_method: method used for imputation, with the following options:
+            'mean', 'median', 'most_frequent', or a string that can be converted to
+            a date or timestamp by `pd.to_datetime` function.
+            If not specified, the default is "1970-01-01" (unix epoch).
+        """
+        if not impute_method:
+            impute_method = self.EPOCH
+        self.impute_method = impute_method
+        if impute_method in ("mean", "median", "most_frequent"):
+            # The `impute_value` will be determined by the `fit` function.
+            self.impute_value = None
+        else:
+            self.impute_value = pd.to_datetime(self.impute_method)
+
+    @abstractmethod
+    def _to_datetime(self, X):
+        """Convert the input to datetime objects (essentially floating numbers)."""
+        pass  # pragma: no cover
+
+    @abstractmethod
+    def _include_timestamp(self):
+        """Whether the output should include timestamp features."""
+        pass  # pragma: no cover
+
     def fit(self, X, y=None):
         """
         Do nothing and return the estimator unchanged.
```

Comment:
Shall we change this docstring? It now calculates the imputation value when calling fit.

Reply:
Then do we want to put everythin into one transformer or create two?

Reply:
Updated the docstring. I'm still leaning towards to keep things in one transformer for now due to technical convenience (see my other comments).

Reply:
Agree on keeping it in one for now. We need the `pd.to_datetime` first, and then do imputation, and finally calculate the datetime features. Seems need to split to at least three classes, which might be an overkill.